src/macos/FSEventsBackend.cc

```diff
@@ -9,6 +9,7 @@
 #include "../Watcher.hh"
 
 #define CONVERT_TIME(ts) ((uint64_t)ts.tv_sec * 1000000000 + ts.tv_nsec)
+#define IGNORED_FLAGS (kFSEventStreamEventFlagItemIsHardlink | kFSEventStreamEventFlagItemIsLastHardlink | kFSEventStreamEventFlagItemIsSymlink | kFSEventStreamEventFlagItemIsDir | kFSEventStreamEventFlagItemIsFile)
```

Comment:
It seems to me that this change could address what https://github.com/parcel-bundler/watcher/pull/57 indicated.

Reply:
I believe we should actually handle those, not ignore them. We already ran into a case in parcel where writing many files quickly produces these flags and means we can't determine what changed accurately. I believe the plan is to emit a specific error code in those cases so the client can handle them. We'll work on that separately.

Reply:
Yeah that makes sense. I was more referring to the extra `stat` call and handling from here:

https://github.com/parcel-bundler/watcher/blob/728b44aeec7801bae9b11e02a7fb3f0bdf94b924/src/macos/FSEventsBackend.cc#L73-L74

In case none of the expected flags are set. I think for the ones mentioned you wouldn't have to do the `stat` call to figure out the kind of event because the flags do not seem to indicate a file change.

Reply:
Yeah for now what happens is it emits a `"create"` event for the root directory. Parcel currently uses that to know that it must re-build the entire project because the changes were actually unknown. When we replace that with a better error code, we can remove that.

Reply:
That makes sense (though VS Code wouldn't know how to handle that actually). 

Just to clarify, you are saying that this call:

https://github.com/parcel-bundler/watcher/blob/cca5ca33483bef44246791ae30efc5d263d14429/src/macos/FSEventsBackend.cc#L76

Is being done whenever an event occurs that is none of the expected ones. But you are saying that the ones I listed in https://github.com/parcel-bundler/watcher/pull/69#discussion_r720780611 will only be emitted for the root?

Reply:
Yeah that's what happens. When events are dropped, macOS emits a single event on the root directory with those flags set. At the moment since they aren't handled specially, it results in a create event because of the stat. In the future we'll emit a specific error code so clients can re-scan appropriately.